cdm/core/src/main/java/ucar/nc2/Variable.java

```diff
@@ -2055,9 +2055,13 @@ public String makeDimensionsString() {
      * Set the dimensions using all anonymous (unshared) dimensions
      *
      * @param shape defines the dimension lengths. must be > 0, or -1 for VLEN
-     * @throws RuntimeException if any shape < 1 and not -1.
+     * @throws InvalidRangeException if any shape < 1 and not -1.
      */
-    public T setDimensionsAnonymous(int[] shape) {
+    public T setDimensionsAnonymous(int[] shape) throws InvalidRangeException {
```

Comment:
The `InvalidRangeException` definitely makes more sense than throwing for instance a `RuntimeException`, but I think changing this function's signature could be considered a breaking change-- since this function is `public` and technically anyone could use it in their code.

Reply:
Ahh yeah....that does make sense. Do you suggest removing the method signature and just throwing the exception? To be fully honest, I'm still trying to get oriented and copying the conventions I've seen elsewhere in the code without knowing the full implications of the changes.

Reply:
I think the only way to accomplish this without changing the method signature is to return `null` (or something) instead of throwing the exception, then in places where you have the `try/catch`, check for `null` instead.

Reply:
Or instead of throwing the `InvalidRangeException` here, you could catch and log `IllegalArgumentException`s. Those come from the `Dimension.makeDimensionsAnon` which calls `Dimension::setLength` if the Dimension length is invalid. I think that does a similar check on the shape as you are doing here.

I guess based on Hailey's comment (that we only want to change the behavior for hdf5 and not bufr) that it is best to do the catching and logging in the `H5headerNew` where this is called (like you are currently doing for the `InvalidRangeException`)